apps/cowswap-frontend/src/legacy/components/AddressInputPanel/index.tsx

```diff
@@ -95,14 +138,30 @@ export function AddressInputPanel({
   onChange: (value: string) => void
 }) {
   const { chainId } = useWalletInfo()
+  const chainInfo = getChainInfo(chainId)
 
   const { address, loading, name } = useENS(value)
+  const [chainWarning, setChainWarning] = useState(false)
```

Comment:
What about using an object the type:

`const [chainPrefixWarning, setChainPrefixWarning] = useState<ChainPrefixWatning | undefined>(false)`

Where `ChainPrefixWatning` is:

```
interface ChainPrefixWatning {
   walletChainId: SupportedChain
   recipientChainId: SupportedChain
}
```

This way you can make a nicer warning message

Reply:
or maybe just save the recipientChainId, as the walletChainId is known

Reply:
check it out now.